### PR TITLE
fix failing tests when using dataProviders

### LINKED ISSUE #1: Getting: TypeError: Cannot read property 'map' of undefined; Because I use dataProviders
Hello there!

The jest runner fails whenever I use a `@dataProvider`. If you are not familiar with a data provider, it is like a `test.each()` in jest, where you write 1 test, but you can give it different sets of data. (See: https://phpunit.de/manual/6.5/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers)

That said, when I look at the `result` variable without dataProviders, it looks like this:

```json
{
  "testsuites": {
    "testsuite": [
      {
        "$": {
          "name": "App\\Tests\\Unit\\TestName",
          "file": "TestName.php",
          "tests": "1",
          "assertions": "12",
          "failures": "0",
          "errors": "0",
          "time": "1.193167"
        },
        "testcase": [
          {
            "$": {
              "name": "testSanitation",
              "class": "TestName",
              "file": "TestName.php",
              "line": "27",
              "assertions": "12",
              "time": "1.193167"
            }
          }
        ]
      }
    ]
  }
}
```

But when I look at the result when I do use data providers, I see the following result:

```json
{
  "testsuites": {
    "testsuite": [
      {
        "$": {
          "name": "TestWithDataProvider",
          "file": "TestWithDataProvider.php",
          "tests": "6",
          "assertions": "6",
          "failures": "0",
          "errors": "0",
          "time": "0.093333"
        },
        "testsuite": [
          {
            "$": {
              "name": "TestWithDataProvider::testNameGoesHere",
              "tests": "6",
              "assertions": "6",
              "failures": "0",
              "errors": "0",
              "time": "0.093333"
            },
            "testcase": [
              {
                "$": {
                  "name": "testNameGoesHere with data set #0",
                  "assertions": "1",
                  "time": "0.031252"
                }
              },
              {
                "$": {
                  "name": "testNameGoesHere with data set #1",
                  "assertions": "1",
                  "time": "0.011367"
                }
              },
              {
                "$": {
                  "name": "testNameGoesHere with data set #2",
                  "assertions": "1",
                  "time": "0.011760"
                }
              },
              {
                "$": {
                  "name": "testNameGoesHere with data set #3",
                  "assertions": "1",
                  "time": "0.011313"
                }
              },
              {
                "$": {
                  "name": "testNameGoesHere with data set #4",
                  "assertions": "1",
                  "time": "0.011672"
                }
              },
              {
                "$": {
                  "name": "testNameGoesHere with data set #5",
                  "assertions": "1",
                  "time": "0.015969"
                }
              }
            ]
          }
        ]
      }
    ]
  }
}
```

The important part here is that there is a test suite inside of the test suite. Which eventually throws `TypeError: Cannot read property 'map' of undefined`.

I am going to submit a PR for this as well. hold on!

### PR DESCRIPTION
**Pull request by RobinMalfait**
This should fix #1 

(I am trying to make the tests work on CI but no luck so far, @olyckne if you could help me out here would be awesome 😅)

It might be that there is still a problem if you have multiple dataProvider tests I think, I am not 100% sure.

I am a bit afraid of the `[0]` part where we only use the 1st item in the array.

### CODE REVIEW

**Comment by gevalo1** [comment]
I added a fix for showing the output when there are multiple dataproviders, see 80666c7.

File: examples/simple/tests/ExampleTest.php [hunk 1]
```diff
@@ -16,4 +15,37 @@ public function testItFails()
...
+
+    /**
+     * @dataProvider exampleDataProvider
+     */
+    public function testWithDataProvider($actual, $expected)
+    {
+        $this->assertSame($actual, $expected);
+    }
+
+    protected function exampleDataProvider()
```
Comment (olyckne) [inline examples/simple/tests/ExampleTest.php]: Dataproviders need to be public
```suggestion
    public function exampleDataProvider()
```

File: examples/simple/tests/ExampleTest.php [hunk 2]
```diff
@@ -16,4 +15,37 @@ public function testItFails()
...
+
+    /**
+     * @dataProvider exampleSecondDataProvider
+     */
+    public function testWithAnotherDataProvider($actual, $expected)
+    {
+        $this->assertSame($actual, $expected);
+    }
+
+    protected function exampleSecondDataProvider()
```
Comment (olyckne) [inline examples/simple/tests/ExampleTest.php]: Dataproviders need to be public
```suggestion
    public function exampleSecondDataProvider()
```

**Review by olyckne — CHANGES_REQUESTED** [review]
Besides the two comments this looks great!

File: examples/simple/tests/ExampleTest.php (hunk 1 above)
Comment (gevalo1) [reply examples/simple/tests/ExampleTest.php]: Oops, missed that. 😅 

Fixed!

**Review by olyckne — APPROVED** [review]

**Comment by olyckne** [comment]
Awesome. Thanks!

**Comment by RobinMalfait** [comment]
Thank you @olyckne!

And thank you @gevalo1 for picking up where I left off!

**Pull request merged**